pkg/summary/summary.go

```diff
@@ -127,10 +132,10 @@ func (s Summarizer) GetSummary() (Summary, error) {
 			Containers:     map[string]containerSummary{},
 		}
 
-		// find all deployments in the namespace for this VPA
-		deployment, err := s.kubeClient.Client.AppsV1().Deployments(namespace).Get(dSummary.DeploymentName, metav1.GetOptions{})
-		if err != nil {
-			klog.Errorf("Error retrieving deployment from API: %v", err)
+		deployment, ok := s.deploymentForVPANamed[vpa.Name]
+		if !ok {
+			klog.Errorf("Error, no matching Deployment found for VPA/%s", vpa.Name)
```

Comment:
klo.Errorf already shows an E for error in the log line. Not need to specify

```suggestion
			klog.Errorf("no matching Deployment found for VPA/%s", vpa.Name)
```